src/transformer/generator.ts

```diff
@@ -33,33 +33,35 @@ export function isZodConstructor(
 }
 
 // #region filter
-type Filter<TSchema extends ZodTypeAny> = (obj: {
+type Filter<TSchema extends ZodTypeAny = ZodTypeAny> = (obj: {
 	def: TSchema['_def'];
 	schema: TSchema;
 	transform: Runner;
 	context: Context;
 }) => boolean;
 // #endregion filter
 
-export type Generator<TSchema extends ZodTypeAny> = (obj: {
+export type Generator<TSchema extends ZodTypeAny = ZodTypeAny> = (obj: {
 	def: TSchema['_def'];
 	schema: TSchema;
 	transform: Runner;
 	context: Context;
 }) => unknown;
 
-export interface Definition<TSchema extends ZodTypeAny> {
-	schema: ZodConstructorOrSchema<TSchema>;
+export interface Definition<TSchema extends ZodTypeAny = ZodTypeAny> {
+	schema?: ZodConstructorOrSchema<TSchema>;
```

Comment:
This needs a test.

I also was thinking about making making one of the props `schema` and `filter` always required.
So that you don't accidentally create a generator that matches every type.
What do you think?

Reply:
I think a valid use case is having a fallback generator for non-matches.

```ts
const FallbackGenerator = Generator({
  output: () => 'fallback'
})

new Fixture().extend([
    // ...defaultGenerators,
    // custom_generator_1,
    // custom_generator_2,
    // custom_generator_3,
    FallbackGenerator
])
```

I'm also not sure we have to enforce "filters" (schema/filter). In most cases, I think a user is building a generator with the intent to filter and will provide some mechanism for doing so out of their nature.

How do you propose we test this?
```ts
export interface Definition<TSchema extends ZodTypeAny = ZodTypeAny> {
	schema?: ZodConstructorOrSchema<TSchema>;
```

Reply:
Ah, I haven't thought of the fallback generator, but I assume that there is something wrong when that one is hit.
Let's leave it like this, we can always enforce this when we notice that problems start to arise (if they arise at all).

We can test this as with other tests, create a fixture and extend it with a generator that uses `filter` (without `schema`). The test expects that that generator is invoked. If you want, this can also be used as an example.